src/runtime_src/core/common/api/xrt_xclbin.cpp

```diff
@@ -523,7 +528,8 @@ class xclbin_full : public xclbin_impl
   std::vector<char> m_axlf;  // complete copy of xclbin raw data
   const axlf* m_top = nullptr;
   uuid m_uuid;
-  std::map<axlf_section_kind, std::vector<char>> m_axlf_sections;
+  //  std::map<axlf_section_kind, std::vector<char>> m_axlf_sections;
+  std::multimap<axlf_section_kind, std::vector<char>> m_axlf_sections;
```

Comment:
This is not obvious.  Why are their multiple blobs for a single section?  Where is this format documented?
Dead code should be removed as opposed to commented out.

I think I am inferring that there are multiple SOFT_KERNEL sections in the xclbin, I guess one per soft kernel.    And as such I understand the get_axlf_section**s** APIs.    For this POC it makes sense, but we need this documented in the device.h API you added an also in xrt_xclbin.* some place.  There also was a newly added get_axlf_section to xrt::xclbin itself, I wonder how this is impacted?